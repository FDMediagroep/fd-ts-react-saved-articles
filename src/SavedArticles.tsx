import React, { PureComponent } from "react";
import Card, { CardTypes, getAllCardStyles } from "@fdmg/fd-card";
import TypoGraphy, { getAllTextStyles } from "@fdmg/fd-typography";
import { createGlobalStyle } from "styled-components";

interface NewsItem {
    id: string;
    dateTime: string;
    label: string;
    url: string;
}

export interface Props {
    cardStyle: CardTypes;
    className?: string;
    items: NewsItem[];
    onDelete?: (id: string) => void;
    title?: string;
}

export default class SavedArticles extends PureComponent<Props, any> {
    state: any = {
        showAll: false
    };

    onDelete = (e: React.MouseEvent<HTMLElement>) => {
        if (this.props.onDelete) { this.props.onDelete(e.currentTarget.getAttribute('data-uuid') as string); }
    }

    onShowMore = () => {
        this.setState({showAll: true});
    }

    onShowLess = () => {
        this.setState({showAll: false});
    }
    render() {
        return (
            <>
                <GlobalStyle/>
                <Card cardStyle={this.props.cardStyle} className={`fd-saved-articles${this.props.className ? ` ${this.props.className}` : ''}${this.state.showAll ? ' show-all' : ''}`}>
                    <TypoGraphy textStyle="card-h" className="fd-saved-articles-h"><h3>{this.props.title ? this.props.title : 'Bewaarde artikelen'}</h3></TypoGraphy>
                    <nav>
                        {
                            this.props.items.map((newsItem, idx: number) => {
                                if (idx > 4) { return null; }
                                return (
                                    <div className="news-item" key={newsItem.id}>
                                        <a href={newsItem.url}>
                                            <time>{newsItem.dateTime}</time>
                                            <span>{newsItem.label}</span>
                                        </a>
                                        <i className="icon-times" data-uuid={newsItem.id} onClick={this.onDelete}/>
                                    </div>
                                );
                            })
                        }
                    </nav>
                    {this.props.items.length > 4 ?
                        <>
                            <nav className="optional-content">
                                {
                                    this.props.items.map((newsItem, idx: number) => {
                                        if (idx <= 4) { return null; }
                                        return (
                                            <div className="news-item" key={newsItem.id}>
                                                <a href={newsItem.url}>
                                                    <time>{newsItem.dateTime}</time>
                                                    <span>{newsItem.label}</span>
                                                </a>
                                                <i className="icon-times" data-uuid={newsItem.id} onClick={this.onDelete}/>
                                            </div>
                                        );
                                    })
                                }
                            </nav>
                            <span className="show-less" onClick={this.onShowLess}><i className="icon-chevron-up"/><div>Toon minder</div></span>
                            <span className="show-more" onClick={this.onShowMore}><div>Toon meer</div><i className="icon-chevron-down"/></span>
                        </> : null}
                </Card>
            </>
        );
    }
}

const GlobalStyle = createGlobalStyle`
.fd-saved-articles {
    .card-h.fd-saved-articles-h,
    .news-item > a {
        padding: 12px 10px;
        @media only screen and (min-width: 641px) {
            padding: 12px;
        }
        @media only screen and (min-width: 861px) {
            padding: 12px 15px;
        }
    }

    .news-item {
        display: flex;
        align-items: stretch;
        border-top: 1px solid rgba(0,0,0,0.1);
        &:hover a {
            color: #49a4a2;
        }
        i {
            display: flex;
            align-items: center;
            justify-content: center;
            min-width: 44px;
            cursor: pointer;
            &:hover {
                background: rgba(0,0,0,0.04);
            }
        }
    }
    a {
        text-decoration: none;
        flex: 1 1 auto;
        font-family: 'ProximaNovaRegular', sans-serif;
        line-height: 1.2em;
        color: #191919;
        font-size: 1rem;
        @media only screen and (min-width: 641px) {
            font-size: 1.0625rem;
        }
    }
    time {
        color: #677381;
        display: block;
        font-size: 13px;
        @media only screen and (min-width: 641px) {
            font-size: 14px;
        }
    }

    i {
        color: #677381;
        display: flex;
        align-content: center;
        justify-content: center;
        user-select: none;
    }

    .show-less,
    .optional-content {
        display: none;
    }

    .show-more {
        display: flex;
    }

    &.show-all {
        .show-less {
            display: flex;
        }
        .optional-content {
            display: block;
        }
        .show-more {
            display: none;
        }
    }

    .show-more, .show-less {
        user-select: none;
        border-top: 1px solid rgba(0,0,0,0.1);
        flex-direction: column;
        justify-content: center;
        cursor: pointer;
        padding: .5rem;
        font-family: 'ProximaNovaRegular',Helvetica,sans-serif;
        font-size: 1rem;
        @media only screen and (min-width: 641px) {
            font-size: 1.0625rem;
        }
        &:hover {
            background: rgba(0, 0, 0, 0.04);
        }
        div {
            text-align: center;
        }
    }
}
`;

export const SavedArticlesStyle = createGlobalStyle`
${getAllCardStyles().globalStyle.rules}
${getAllTextStyles(['card-h']).globalStyle.rules}
${(GlobalStyle as any).globalStyle.rules}
`;
