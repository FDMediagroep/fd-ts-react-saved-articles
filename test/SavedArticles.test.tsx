import TestRenderer from 'react-test-renderer';
import SavedArticles from '../src/SavedArticles';
import React from 'react';

describe('SavedArticles', () => {
    const news = [{
        id: '1',
        dateTime: `11:18`,
        label: 'Trump vraagt om politieke samenwerking in State of the Union',
        url: 'https://fd.nl/economie-politiek/1288267/trump-vraagt-om-politieke-samenwerking-in-state-of-the-union',
    }, {
        id: '2',
        dateTime: `12:19`,
        label: 'Leraren slaan alarm: scholen stoppen met informatica in de bovenbouw',
        url: 'https://fd.nl/economie-politiek/1287621/leraren-slaan-alarm-scholen-stoppen-met-informatica-in-de-bovenbouw'
    }, {
        id: '3',
        dateTime: `13:20`,
        label: 'Verschillen tussen eurolanden gaan weer opspelen ',
        url: 'https://fd.nl/achtergrond/1288289/verschillen-tussen-eurolanden-gaan-weer-opspelen'
    }, {
        id: '4',
        dateTime: `14:21`,
        label: 'ING boekt meer winst dan verwacht',
        url: 'https://fd.nl/ondernemen/1288060/ing-boekt-meer-winst-dan-verwacht'
    }, {
        id: '5',
        dateTime: `Vandaag`,
        label: 'ING blinkt uit op hoger Damrak',
        url: 'https://fd.nl/beurs/1288359/beleggers-reageren-op-cijfers-ing-en-tomtom'
    }, {
        id: '6',
        dateTime: `Gisteren`,
        label: 'Vertrouwen van werknemers is kostbaar goed dat menige werkgever keer op keer blijft beschamen',
        url: 'https://fd.nl/opinie/1288043/vertrouwen-van-werknemers-is-kostbaar-goed-dat-menige-werkgever-keer-op-keer-blijft-beschamen'
    }, {
        id: '7',
        dateTime: `Eergisteren`,
        label: 'Verschillen tussen eurolanden gaan weer opspelen',
        url: 'https://fd.nl/achtergrond/1288289/verschillen-tussen-eurolanden-gaan-weer-opspelen'
    }, {
        id: '8',
        dateTime: `Vorige week`,
        label: 'Politieke druk op Vestager is onverteerbaar',
        url: 'https://fd.nl/economie-politiek/1288325/politieke-druk-op-vestager-is-onverteerbaar'
    }, {
        id: '9',
        dateTime: `Afgelopen maand`,
        label: 'Leraren slaan alarm: scholen stoppen met informatica in de bovenbouw',
        url: 'https://fd.nl/economie-politiek/1287621/leraren-slaan-alarm-scholen-stoppen-met-informatica-in-de-bovenbouw'
    }, {
        id: '10',
        dateTime: `Vorig jaar`,
        label: 'Zorgsector kan kunstmatige intelligentie veel slimmer inzetten',
        url: 'https://fd.nl/opinie/1288142/zorgsector-kan-kunstmatige-intelligentie-veel-slimmer-inzetten'
    }];

    test('renders correctly', () => {
        let savedArticles = TestRenderer.create(<SavedArticles
            cardStyle="default"
            items={news}
            onDelete={() => {}}
        />);
        expect(savedArticles.toJSON()).toMatchSnapshot();

        savedArticles = TestRenderer.create(<SavedArticles
            cardStyle="default"
            items={news}
            onDelete={() => {}}
        />);
        expect(savedArticles.toJSON()).toMatchSnapshot();
    });

    test('renders persoonlijk correctly', () => {
        let savedArticles = TestRenderer.create(<SavedArticles
            cardStyle="persoonlijk"
            items={news}
            onDelete={() => {}}
        />);
        expect(savedArticles.toJSON()).toMatchSnapshot();

        savedArticles = TestRenderer.create(<SavedArticles
            cardStyle="persoonlijk"
            items={news}
            onDelete={() => {}}
        />);
        expect(savedArticles.toJSON()).toMatchSnapshot();
    });
});
