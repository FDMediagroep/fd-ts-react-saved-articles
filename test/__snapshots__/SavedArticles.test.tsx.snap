// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedArticles renders correctly 1`] = `
<div
  cardStyle="default"
  className="fd-card fd-saved-articles default"
>
  <h3
    className="fd-typography card-h fd-saved-articles-h"
    textStyle="card-h"
  >
    Bewaarde artikelen
  </h3>
  <nav>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/economie-politiek/1288267/trump-vraagt-om-politieke-samenwerking-in-state-of-the-union"
      >
        <time>
          11:18
        </time>
        <span>
          Trump vraagt om politieke samenwerking in State of the Union
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="1"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/economie-politiek/1287621/leraren-slaan-alarm-scholen-stoppen-met-informatica-in-de-bovenbouw"
      >
        <time>
          12:19
        </time>
        <span>
          Leraren slaan alarm: scholen stoppen met informatica in de bovenbouw
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="2"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/achtergrond/1288289/verschillen-tussen-eurolanden-gaan-weer-opspelen"
      >
        <time>
          13:20
        </time>
        <span>
          Verschillen tussen eurolanden gaan weer opspelen 
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="3"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/ondernemen/1288060/ing-boekt-meer-winst-dan-verwacht"
      >
        <time>
          14:21
        </time>
        <span>
          ING boekt meer winst dan verwacht
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="4"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/beurs/1288359/beleggers-reageren-op-cijfers-ing-en-tomtom"
      >
        <time>
          Vandaag
        </time>
        <span>
          ING blinkt uit op hoger Damrak
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="5"
        onClick={[Function]}
      />
    </div>
  </nav>
  <nav
    className="optional-content"
  >
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/opinie/1288043/vertrouwen-van-werknemers-is-kostbaar-goed-dat-menige-werkgever-keer-op-keer-blijft-beschamen"
      >
        <time>
          Gisteren
        </time>
        <span>
          Vertrouwen van werknemers is kostbaar goed dat menige werkgever keer op keer blijft beschamen
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="6"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/achtergrond/1288289/verschillen-tussen-eurolanden-gaan-weer-opspelen"
      >
        <time>
          Eergisteren
        </time>
        <span>
          Verschillen tussen eurolanden gaan weer opspelen
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="7"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/economie-politiek/1288325/politieke-druk-op-vestager-is-onverteerbaar"
      >
        <time>
          Vorige week
        </time>
        <span>
          Politieke druk op Vestager is onverteerbaar
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="8"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/economie-politiek/1287621/leraren-slaan-alarm-scholen-stoppen-met-informatica-in-de-bovenbouw"
      >
        <time>
          Afgelopen maand
        </time>
        <span>
          Leraren slaan alarm: scholen stoppen met informatica in de bovenbouw
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="9"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/opinie/1288142/zorgsector-kan-kunstmatige-intelligentie-veel-slimmer-inzetten"
      >
        <time>
          Vorig jaar
        </time>
        <span>
          Zorgsector kan kunstmatige intelligentie veel slimmer inzetten
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="10"
        onClick={[Function]}
      />
    </div>
  </nav>
  <span
    className="show-less"
    onClick={[Function]}
  >
    <i
      className="icon-chevron-up"
    />
    <div>
      Toon minder
    </div>
  </span>
  <span
    className="show-more"
    onClick={[Function]}
  >
    <div>
      Toon meer
    </div>
    <i
      className="icon-chevron-down"
    />
  </span>
</div>
`;

exports[`SavedArticles renders correctly 2`] = `
<div
  cardStyle="default"
  className="fd-card fd-saved-articles default"
>
  <h3
    className="fd-typography card-h fd-saved-articles-h"
    textStyle="card-h"
  >
    Bewaarde artikelen
  </h3>
  <nav>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/economie-politiek/1288267/trump-vraagt-om-politieke-samenwerking-in-state-of-the-union"
      >
        <time>
          11:18
        </time>
        <span>
          Trump vraagt om politieke samenwerking in State of the Union
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="1"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/economie-politiek/1287621/leraren-slaan-alarm-scholen-stoppen-met-informatica-in-de-bovenbouw"
      >
        <time>
          12:19
        </time>
        <span>
          Leraren slaan alarm: scholen stoppen met informatica in de bovenbouw
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="2"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/achtergrond/1288289/verschillen-tussen-eurolanden-gaan-weer-opspelen"
      >
        <time>
          13:20
        </time>
        <span>
          Verschillen tussen eurolanden gaan weer opspelen 
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="3"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/ondernemen/1288060/ing-boekt-meer-winst-dan-verwacht"
      >
        <time>
          14:21
        </time>
        <span>
          ING boekt meer winst dan verwacht
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="4"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/beurs/1288359/beleggers-reageren-op-cijfers-ing-en-tomtom"
      >
        <time>
          Vandaag
        </time>
        <span>
          ING blinkt uit op hoger Damrak
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="5"
        onClick={[Function]}
      />
    </div>
  </nav>
  <nav
    className="optional-content"
  >
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/opinie/1288043/vertrouwen-van-werknemers-is-kostbaar-goed-dat-menige-werkgever-keer-op-keer-blijft-beschamen"
      >
        <time>
          Gisteren
        </time>
        <span>
          Vertrouwen van werknemers is kostbaar goed dat menige werkgever keer op keer blijft beschamen
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="6"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/achtergrond/1288289/verschillen-tussen-eurolanden-gaan-weer-opspelen"
      >
        <time>
          Eergisteren
        </time>
        <span>
          Verschillen tussen eurolanden gaan weer opspelen
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="7"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/economie-politiek/1288325/politieke-druk-op-vestager-is-onverteerbaar"
      >
        <time>
          Vorige week
        </time>
        <span>
          Politieke druk op Vestager is onverteerbaar
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="8"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/economie-politiek/1287621/leraren-slaan-alarm-scholen-stoppen-met-informatica-in-de-bovenbouw"
      >
        <time>
          Afgelopen maand
        </time>
        <span>
          Leraren slaan alarm: scholen stoppen met informatica in de bovenbouw
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="9"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/opinie/1288142/zorgsector-kan-kunstmatige-intelligentie-veel-slimmer-inzetten"
      >
        <time>
          Vorig jaar
        </time>
        <span>
          Zorgsector kan kunstmatige intelligentie veel slimmer inzetten
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="10"
        onClick={[Function]}
      />
    </div>
  </nav>
  <span
    className="show-less"
    onClick={[Function]}
  >
    <i
      className="icon-chevron-up"
    />
    <div>
      Toon minder
    </div>
  </span>
  <span
    className="show-more"
    onClick={[Function]}
  >
    <div>
      Toon meer
    </div>
    <i
      className="icon-chevron-down"
    />
  </span>
</div>
`;

exports[`SavedArticles renders persoonlijk correctly 1`] = `
<div
  cardStyle="persoonlijk"
  className="fd-card fd-saved-articles persoonlijk"
>
  <h3
    className="fd-typography card-h fd-saved-articles-h"
    textStyle="card-h"
  >
    Bewaarde artikelen
  </h3>
  <nav>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/economie-politiek/1288267/trump-vraagt-om-politieke-samenwerking-in-state-of-the-union"
      >
        <time>
          11:18
        </time>
        <span>
          Trump vraagt om politieke samenwerking in State of the Union
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="1"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/economie-politiek/1287621/leraren-slaan-alarm-scholen-stoppen-met-informatica-in-de-bovenbouw"
      >
        <time>
          12:19
        </time>
        <span>
          Leraren slaan alarm: scholen stoppen met informatica in de bovenbouw
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="2"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/achtergrond/1288289/verschillen-tussen-eurolanden-gaan-weer-opspelen"
      >
        <time>
          13:20
        </time>
        <span>
          Verschillen tussen eurolanden gaan weer opspelen 
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="3"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/ondernemen/1288060/ing-boekt-meer-winst-dan-verwacht"
      >
        <time>
          14:21
        </time>
        <span>
          ING boekt meer winst dan verwacht
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="4"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/beurs/1288359/beleggers-reageren-op-cijfers-ing-en-tomtom"
      >
        <time>
          Vandaag
        </time>
        <span>
          ING blinkt uit op hoger Damrak
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="5"
        onClick={[Function]}
      />
    </div>
  </nav>
  <nav
    className="optional-content"
  >
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/opinie/1288043/vertrouwen-van-werknemers-is-kostbaar-goed-dat-menige-werkgever-keer-op-keer-blijft-beschamen"
      >
        <time>
          Gisteren
        </time>
        <span>
          Vertrouwen van werknemers is kostbaar goed dat menige werkgever keer op keer blijft beschamen
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="6"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/achtergrond/1288289/verschillen-tussen-eurolanden-gaan-weer-opspelen"
      >
        <time>
          Eergisteren
        </time>
        <span>
          Verschillen tussen eurolanden gaan weer opspelen
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="7"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/economie-politiek/1288325/politieke-druk-op-vestager-is-onverteerbaar"
      >
        <time>
          Vorige week
        </time>
        <span>
          Politieke druk op Vestager is onverteerbaar
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="8"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/economie-politiek/1287621/leraren-slaan-alarm-scholen-stoppen-met-informatica-in-de-bovenbouw"
      >
        <time>
          Afgelopen maand
        </time>
        <span>
          Leraren slaan alarm: scholen stoppen met informatica in de bovenbouw
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="9"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/opinie/1288142/zorgsector-kan-kunstmatige-intelligentie-veel-slimmer-inzetten"
      >
        <time>
          Vorig jaar
        </time>
        <span>
          Zorgsector kan kunstmatige intelligentie veel slimmer inzetten
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="10"
        onClick={[Function]}
      />
    </div>
  </nav>
  <span
    className="show-less"
    onClick={[Function]}
  >
    <i
      className="icon-chevron-up"
    />
    <div>
      Toon minder
    </div>
  </span>
  <span
    className="show-more"
    onClick={[Function]}
  >
    <div>
      Toon meer
    </div>
    <i
      className="icon-chevron-down"
    />
  </span>
</div>
`;

exports[`SavedArticles renders persoonlijk correctly 2`] = `
<div
  cardStyle="persoonlijk"
  className="fd-card fd-saved-articles persoonlijk"
>
  <h3
    className="fd-typography card-h fd-saved-articles-h"
    textStyle="card-h"
  >
    Bewaarde artikelen
  </h3>
  <nav>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/economie-politiek/1288267/trump-vraagt-om-politieke-samenwerking-in-state-of-the-union"
      >
        <time>
          11:18
        </time>
        <span>
          Trump vraagt om politieke samenwerking in State of the Union
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="1"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/economie-politiek/1287621/leraren-slaan-alarm-scholen-stoppen-met-informatica-in-de-bovenbouw"
      >
        <time>
          12:19
        </time>
        <span>
          Leraren slaan alarm: scholen stoppen met informatica in de bovenbouw
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="2"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/achtergrond/1288289/verschillen-tussen-eurolanden-gaan-weer-opspelen"
      >
        <time>
          13:20
        </time>
        <span>
          Verschillen tussen eurolanden gaan weer opspelen 
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="3"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/ondernemen/1288060/ing-boekt-meer-winst-dan-verwacht"
      >
        <time>
          14:21
        </time>
        <span>
          ING boekt meer winst dan verwacht
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="4"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/beurs/1288359/beleggers-reageren-op-cijfers-ing-en-tomtom"
      >
        <time>
          Vandaag
        </time>
        <span>
          ING blinkt uit op hoger Damrak
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="5"
        onClick={[Function]}
      />
    </div>
  </nav>
  <nav
    className="optional-content"
  >
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/opinie/1288043/vertrouwen-van-werknemers-is-kostbaar-goed-dat-menige-werkgever-keer-op-keer-blijft-beschamen"
      >
        <time>
          Gisteren
        </time>
        <span>
          Vertrouwen van werknemers is kostbaar goed dat menige werkgever keer op keer blijft beschamen
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="6"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/achtergrond/1288289/verschillen-tussen-eurolanden-gaan-weer-opspelen"
      >
        <time>
          Eergisteren
        </time>
        <span>
          Verschillen tussen eurolanden gaan weer opspelen
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="7"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/economie-politiek/1288325/politieke-druk-op-vestager-is-onverteerbaar"
      >
        <time>
          Vorige week
        </time>
        <span>
          Politieke druk op Vestager is onverteerbaar
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="8"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/economie-politiek/1287621/leraren-slaan-alarm-scholen-stoppen-met-informatica-in-de-bovenbouw"
      >
        <time>
          Afgelopen maand
        </time>
        <span>
          Leraren slaan alarm: scholen stoppen met informatica in de bovenbouw
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="9"
        onClick={[Function]}
      />
    </div>
    <div
      className="news-item"
    >
      <a
        href="https://fd.nl/opinie/1288142/zorgsector-kan-kunstmatige-intelligentie-veel-slimmer-inzetten"
      >
        <time>
          Vorig jaar
        </time>
        <span>
          Zorgsector kan kunstmatige intelligentie veel slimmer inzetten
        </span>
      </a>
      <i
        className="icon-times"
        data-uuid="10"
        onClick={[Function]}
      />
    </div>
  </nav>
  <span
    className="show-less"
    onClick={[Function]}
  >
    <i
      className="icon-chevron-up"
    />
    <div>
      Toon minder
    </div>
  </span>
  <span
    className="show-more"
    onClick={[Function]}
  >
    <div>
      Toon meer
    </div>
    <i
      className="icon-chevron-down"
    />
  </span>
</div>
`;
